core/base/src/utils/mapping.ts

```diff
@@ -211,12 +211,16 @@ const toMapping = <
     throw new Error("Invalid mapping: empty");
 
   const definedShape = (shape === undefined)
-    ? [range(crr[0].length - 1), [crr[0].length - 1]]
-    : shape.map(ind => typeof ind === "number" ? [ind] : ind);
+    ? [range(crr[0].length - 1), crr[0].length - 1]
+    : shape
+  const adjustedShape = definedShape.map(ind => typeof ind === "number" ? [ind] : ind);
 
+  let finalSingleton = typeof definedShape[1] === 'number';
```

Comment:
if the second entry in the requested shape is a number, expect it to be a single value